### PR TITLE
Navigation service

### PR DESCRIPTION
**Pull request by bex1**
Contains visibility refactoring and constructor style refactoring.

Removes unused components.

The route files concerned are JARoutes, JAroute, app.routing.module, NavigationService.

### CODE REVIEW

File: src/app/services/api-call.service.ts [hunk 1]
```diff
@@ -100,15 +112,17 @@ export class ApiCall {
 
   private handleResponseErrors(response) {
     if (response.status === 401) {
-      let tokenExpiredObject = _.find(response.json().errors, {code: 'token_expired'});
+      let tokenExpiredObject = _.find(response.json().errors, { code: 'token_expired' });
       if (!!tokenExpiredObject) {
         this.userManager.deleteUser();
-        this.router.navigate(['/login']);
+        // TODO should be JARoutes.login . Will not load if used BUG
```
Comment (Ekman1991) [inline src/app/services/api-call.service.ts:118]: Open issue for this instead of comment?

File: src/app/services/api-call.service.ts [hunk 2]
```diff
@@ -100,15 +112,17 @@ export class ApiCall {
...
+        this.navigationService.navigate({ url: () => 'login'});
       }
     }
 
     if (response.status === 0 || response.status === 400 || response.status >= 500) {
-      this.router.navigateByUrl('/error/' + response.status, { skipLocationChange: true });
+      // TODO should be JARoutes.error . Will not load if used BUG
```
Comment (Ekman1991) [inline src/app/services/api-call.service.ts:124]: Open issue for this instead of comment?

File: src/app/app.component.ts [hunk 3]
```diff
@@ -22,98 +17,79 @@ import {SystemLanguagesService} from './services/system-languages.service';
...
     this.authManager.getUserChangeEmmiter().subscribe(user => {
-      this.isCompanyUser = this.userManager.isCompanyUser();
       this.user = user;
     });
 
     this.userManager.getUserChangeEmmiter().subscribe(user => {
       this.user = user;
     });
 
+    console.log('Application component initialized ...');
```
Comment (bellevik) [inline src/app/app.component.ts]: Do we need this log?

**Review by bellevik — APPROVED** [review]
👍

File: src/app/services/proxy/job-proxy.service.ts [hunk 4]
```diff
@@ -19,62 +21,62 @@ export class JobProxy {
...
     return this.apiCall.get('jobs/' + jobId + '/users/' + userJobId, additionOptions).then(response => {
       return new UserJob(response.data);
     });
   }
 
-  addRating(jobId, ratingData) {
+  public addRating(jobId, ratingData) {
     return this.apiCall.post('jobs/' + jobId + '/ratings', ratingData);
     //   .then(response => {
     //   return; //TODO: rating Entity
```
Comment (Ekman1991) [inline src/app/services/proxy/job-proxy.service.ts]: Open issue?

File: src/app/services/api-call.service.ts (hunk 1 above)
Comment (bex1) [reply src/app/services/api-call.service.ts:118]: will do

File: src/app/services/api-call.service.ts (hunk 2 above)
Comment (bex1) [reply src/app/services/api-call.service.ts:124]: will do

**Pull request merged**